src/scripts/main.js

```diff
@@ -1,5 +1,26 @@
 'use strict';
 
+const spider = document.querySelector('.spider');
+const wall = document.querySelector('.wall');
+
 document.addEventListener('click', (e) => {
-  // write code here
+  const field = e.target.closest('.wall');
+
+  if (!field) {
+    return;
+  }
+
+  const wallRect = wall.getBoundingClientRect();
+  const spiderRect = spider.getBoundingClientRect();
+  const widthSpider = spider.offsetWidth;
+  const border = wall.clientTop;
+
+  let x = e.clientX - widthSpider / 2 - border - wallRect.left;
+  let y = e.clientY - widthSpider / 2 - border - wallRect.top;
+
+  x = Math.max(0, Math.min(x, wall.clientWidth - spiderRect.width));
+  y = Math.max(0, Math.min(y, wall.clientHeight - spiderRect.height));
+
+  spider.style.left = x + 'px';
+  spider.style.top = y + 'px';
 });
```

Comment:
It's a good practice to separate your code into smaller, more manageable functions. This function is handling multiple tasks (getting the clicked field, calculating the spider's new position, and moving the spider), so it could be separated into smaller functions. This would improve the readability and testability of your code.